sway-lsp/src/capabilities/text_sync.rs

```diff
@@ -21,35 +21,31 @@ pub fn handle_open_file(
         }
     }
 
-    match session.parse_document(path) {
-        Ok(diagnostics) => diagnostics,
-        Err(DocumentError::FailedToParse(diagnostics)) => diagnostics,
-        _ => vec![],
-    }
+    parse_document(session, path)
 }
 
 pub fn handle_change_file(
     session: Arc<Session>,
     params: DidChangeTextDocumentParams,
-) -> Result<(), DocumentError> {
-    session.update_text_document(&params.text_document.uri, params.content_changes)
+) -> Vec<Diagnostic> {
+    let path = params.text_document.uri.path();
+    let _ = session.update_text_document(&params.text_document.uri, params.content_changes);
```

Comment:
Looks like a `Result` might be getting ignored here now?

Perhaps the return type of this method should be:

```rust
-> Result<Vec<Diagnostic>, DocumentError>
```

And this line should be:

```rust
session.update_text_document(&params.text_document.uri, params.content_changes)?;
```

Otherwise if there's a reason it's ignored, we should add a comment to justify it.

Reply:
That's a fair point. The original possible error comes from using the path as the key into a HashMap of files, if the key doesn't exist an error is returned. The error is being ignored because we never want our server to crash. There is probably no point in returning an error at all then from the `update_text_document` fn. I'll update that fn now.